aptos-move/move-examples/tic-tac-toe/sources/TicTacToe.move

```diff
@@ -0,0 +1,696 @@
+module tic_tac_toe::ttt {
+    use std::vector;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::error;
+    use aptos_framework::account;
+    use aptos_framework::event;
+
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+
+
+
+    //// GAME CONSTANTS
+    const GRID_SIZE: u64 = 9;
+    const WIDTH_AND_HEIGHT: u64 = 3;
+    const EMPTY_CELL: u64 = 3;
+
+    //// PLAYER CONSTANTS
+    const PLAYER_X_TYPE: u64 = 1;
+    const PLAYER_O_TYPE: u64 = 2;
+
+    //// ERROR CODES
+    /// Placing a move on a (x,y) position that is already occupied
+    const EINVALID_MOVE: u64 = 0;
+    /// An address trying to claim a player that is already taken
+    const EPLAYER_TAKEN: u64 = 1;
+    /// Trying to destroy game before game has been finished
+    const EGAME_NOT_DONE: u64 = 2;
+    /// An address can only join as one player, not two
+    const ECANNOT_JOIN_AS_TWO_PLAYERS: u64 = 3;
+    /// A (x,y) move is out of bounds of the 3x3 grid
+    const EOUT_OF_BOUNDS_MOVE: u64 = 4;
+    /// Address doesn't exist in the current Game
+    const EPLAYER_NOT_IN_GAME: u64 = 5;
+    /// An attempt to place moves even though a Game is over
+    const EGAME_HAS_ALREADY_FINISHED: u64 = 6;
+    /// User tries to make two games
+    const EGAME_ALREADY_EXISTS_FOR_USER: u64 = 7;
+    /// Game doesn't exist under address
+    const EGAME_DOESNT_EXIST: u64 = 8;
+    /// Out of turn move or player not in game
+    const EOUT_OF_TURN_MOVE: u64 = 9;
+
+
+    struct GameOverEvent has drop, store {
+        is_game_over: bool,
+    }
+
+    struct Player has copy, drop, store {
+        type: u64,
+        owner: address,
+    }
+
+    struct Board has drop, store {
+        vec: vector<u64>,
+        row: u64,
+        col: u64,
+    }
+
+    struct Game has key, store {
+        board: Board,
+        player_x: Option<Player>,
+        player_o: Option<Player>,
+        is_player_x_turn: bool,
+        is_game_over: bool,
+        game_over_events: event::EventHandle<GameOverEvent>,
+    }
+
+    /*
+     * @notice initializes a valid, playable Game
+     * @dev stores the Game into global storage
+     */
+    public entry fun start_game(creator: &signer) {
+        // check game doesn't already exist under creator address
+        assert!(!exists<Game>(signer::address_of(creator)), error::already_exists(EGAME_ALREADY_EXISTS_FOR_USER));
+        let game = initalize_game(creator);
+        let creator_addr = signer::address_of(creator);
+        choose_player_x(&mut game, creator_addr);
+        move_to<Game>(creator, game);
```

Comment:
This is an example so probably not a big deal, but I'd not structure the game data as a temporary state inside the creator's account. It's better practice to create the game as an object (with its own address), then store it a pointer to the active game in the creator account if I really want to make everything discoverable on chain. Otherwise, such a pointer is not needed if the UI component can rely on indexers to find active games for a player